### PR TITLE
Allow delivery policy of DeliverLastPerSubject, which is helpful for K/V on JetStream

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: server/consumer.go [hunk 1]
```diff
@@ -2953,18 +3031,28 @@ func (o *consumer) setInitialPendingAndStart() {
 		}
 	} else {
 		// Here we are filtered.
-		ss := mset.store.FilteredState(o.sseq, o.cfg.FilterSubject)
-		if ss.Msgs > 0 {
+		dp := o.cfg.DeliverPolicy
+		if dp == DeliverLastPerSubject && o.hasSkipListPending() && o.sseq < o.lss.resume {
+			if o.lss != nil {
```
Comment (ripienaar) [inline server/consumer.go:3036]: i tink `o.hasSkipListPending()` check above implies o.lss will be non nil

Comment (derekcollison) [reply server/consumer.go:3036]: Upper clause has an additional conditional so I need to check again.

```go
if dp == DeliverLastPerSubject && o.hasSkipListPending() && o.sseq < o.lss.resume {
```

**Review by ripienaar — APPROVED** [review]
LGTM

**Comment by aricart** [comment]
@derekcollison - the next ask is probably to do last_per_subject starting at a known sequence. This would allow a client to store a 'stop' location in the stream, but only have to review the latest changes to some of the data.

**Comment by derekcollison** [comment]
Good idea to discuss between you, R.I. and others.

**Pull request merged**